mcs/class/corlib/System/MonoCustomAttrs.cs

```diff
@@ -301,6 +301,35 @@ internal static IList<CustomAttributeData> GetCustomAttributesData (ICustomAttri
 				throw new ArgumentNullException ("obj");
 
 			CustomAttributeData [] attrs = GetCustomAttributesDataInternal (obj);
+
+			object [] pseudoAttrs = System.Array.FindAll (GetPseudoCustomAttributes (obj, null), attr => attr is MarshalAsAttribute);
+			if (pseudoAttrs.Length > 0)
+			{
+				CustomAttributeData [] maAttrsData = new CustomAttributeData [pseudoAttrs.Length];
+				for (var i = 0; i < pseudoAttrs.Length; i++) {
+					MarshalAsAttribute maAttr = (MarshalAsAttribute)pseudoAttrs[i];
+					var attrValue = maAttr.Value;
+					var customAttributeData = (CustomAttributeData)Activator.CreateInstance (typeof (CustomAttributeData), true);
```

Comment:
Maybe add a static method to `CustomAttributeData.CreateFromPseudoCustomAttr()` that initializes a `CustomAttributeData` instance directly with the ctorArgs and ctorInfo?  Then you wouldn't have to jump through reflection.